lib/liquid/tags/render.rb

```diff
@@ -58,7 +60,7 @@ def render_tag(context, output)
       }
 
       variable = @variable_name_expr ? context.evaluate(@variable_name_expr) : nil
-      if variable.is_a?(Array)
+      if variable.respond_to?(:each) && variable.respond_to?(:count) && @loop_variable
```

Comment:
`count` is part of `Enumerable` so Hash-like drops would still render incorrectly.

How about `variable.respond_to?(:each) && !variable.respond_to?(:keys)`?

Also we could move the `@loop_variable` check to the front since it's a fast check.

Reply:
In this implementation, It doesn't mind about `Hash` or `Hash-like Drop` as the idea is that the `for` has been explicitly set. 

`variable.respond_to?(:each) && !variable.respond_to?(:keys)` could be a variation to this approach and means `with` and `for` could stay the same

Reply:
Ah of course. In that case this is fine then. The developer's intent was to iterate, so even if the argument is `Hash`-like, iterate we should.

Reply:
:+1 on the `if` shortcutting, will change now